internal/collector/iis/iis_app_pool_was.go

```diff
@@ -168,6 +168,10 @@ func (c *Collector) buildAppPoolWAS() error {
 }
 
 func (c *Collector) collectAppPoolWAS(ch chan<- prometheus.Metric) error {
+	if c.perfDataCollectorAppPoolWAS == nil {
```

Comment:
Whats the reason for this?

Reply:
There is already a null check on function side, with is sufficient. It may not make sense to have the nil check twice

https://github.com/prometheus-community/windows_exporter/blob/370a42b79ad518695a7391b94cc49c12b13dc63f/internal/pdh/collector.go#L279-L281